### PR TITLE
Implement some indexapi methods, class for managing an index

### PR DESCRIPTION
**Pull request by tommyjcarpenter**

### CODE REVIEW

File: merqube_client_lib/api_client/merqube_client.py [hunk 1]
```diff
@@ -3,15 +3,107 @@
...
+        """
+        return Index.parse_obj(self.get_index_manifest())
+
+    def get_returns(
+        self,
+        returns_metric: str = "price_return",
+        start_date: pd.Timestamp | None = None,
+        end_date: pd.Timestamp | None = None,
+    ) -> pd.DataFrame:
+        raise NotImplementedError((returns_metric, start_date, end_date))
```
Comment (tommyjcarpenter) [inline merqube_client_lib/api_client/merqube_client.py:54]: to come

File: tests/unit/test_secapi_client_reads.py [hunk 2]
```diff
@@ -32,6 +32,14 @@
 )
 
 
+def test_public_client():
```
Comment (tommyjcarpenter) [inline tests/unit/test_secapi_client_reads.py:35]: moved

File: tests/integration/test_indexapi_client_single.py [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+from merqube_client_lib.exceptions import APIError
+from merqube_client_lib.session import get_public_merqube_session
+
+
+def test_nonexistent():
+    with pytest.raises(APIError):
+        MerqubeAPIClientSingleIndex(index_name="thisdoesnotexist", user_session=get_public_merqube_session())
+
+
+def test_single():
```
Comment (aschmied) [inline tests/integration/test_indexapi_client_single.py:14]: 👍

File: merqube_client_lib/api_client/merqube_client.py [hunk 4]
```diff
@@ -3,15 +3,107 @@
...
 class MerqubeAPIClient(IndexAPIClient, SecAPIClient):
-    """combined Merqube client"""
+    """
+    Combined API Client for indexapi + secapi, for methods that do not pertain to a particular index
+    (general queries, CRUD of indices, etc)
+    For a client pertaining to a specific existing index, use MerqubeAPIClientSingleIndex
+    """
+
+
+class MerqubeAPIClientSingleIndex(MerqubeApiClientBase):
```
Comment (aschmied) [inline merqube_client_lib/api_client/merqube_client.py:25]: Wondering about the expected usage pattern here. Is this based on our current internal usage patterns or expected client usage patterns?

How will these clients hang together? Will `MerqubeAPIClient` have a `get_index_client` that returns a `MerqubeAPIClientSingleIndex`? Or will we need to get the name and build `MerqubeAPIClientSingleIndex` manually?

**Review by aschmied — APPROVED** [review]

File: merqube_client_lib/api_client/merqube_client.py (hunk 4 above)
Comment (tommyjcarpenter) [reply merqube_client_lib/api_client/merqube_client.py:25]: this is a good question, not quite sure. 

my thinking was:
- some operations are index-agnostic. bulk secapi queries, and index crud (eg creating a new index) for example
- some operations are for "my [customer] index" - getting returns, other metrics, etc. you obv have to tell it what index you want, 

I was thinking the first is served by APIClient and the latter is served by SingleIndex

however, I am definitely open to a different organization.

or, maybe SingleIndex should inherit from APIClient?

WDYT?

Comment (aschmied) [reply merqube_client_lib/api_client/merqube_client.py:25]: I think it sounds reasonable. Even if a client has multiple indices, they will modify them one at a time (unless we support bulk updates). Having SingleIndex in its own object will make it easier to understand from looking at the API what methods apply to a single index

Comment (tommyjcarpenter) [reply merqube_client_lib/api_client/merqube_client.py:25]: thanks - ill add some docs about this in the coming pr.

**Pull request merged**